File: leetcode/py/136-single-number.py
class Solution:
    def singleNumber(self, nums):
        vals = {}
        for i in nums:
            if i not in vals:
                vals[i] = 1
            else:
                vals[i] += 1
        for j in vals.keys():
            if vals[j] == 1:
                return j
        return -1

File: leetcode/js/136-single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let dict = {};
    for (let i of nums) {
        console.log(i)
        if (!(i in dict)) {
            dict[i] = 1;
        }
        else {
            dict[i] += 1;
        }
    }
    for (let key in dict) {
        if (dict[key] === 1) {
            return key
        }
    }
    return 0;
};